### PR TITLE
MES-7477 / Upgrade packages and Fix vulnerability

### PR DESCRIPTION
**Pull request by matthew2564**
## Description
- Upgrade ```Angular``` (Starting a fresh Ionic project as of 14/12/21 uses Angular 12.x - Followed this angular migration guide to move between versions https://update.angular.io/?l=3&v=10.2-11.0) 
- Upgrade ```NGRX``` to align with Angular version
- Upgrade ```TypeScript``` to 4.x
- Exclude any file ending in ```.mock``` / ```.spec``` or inside the ```angular-mocks```/```ionic-mock``` files from the app build
- Remove un-needed packages such as ```Protractor```,  ```Joi``` and ```enjoi```
- Move dev packages into dev dependencies

## Checklist

- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: buildScripts/crypto-patch.js [hunk 1]
```diff
@@ -1,15 +1,15 @@
-const fs = require('fs');
```
Comment (matthew2564) [inline buildScripts/crypto-patch.js:1]: I think this file can be deleted, need someone to verify GA still working.

File: src/app/pages/journal/components/journal-navigation/journal-navigation.scss [hunk 2]
```diff
@@ -1,6 +1,6 @@
-> div {
-  margin: 12px 0 24px;
-}
+//> div {
```
Comment (matthew2564) [inline src/app/pages/journal/components/journal-navigation/journal-navigation.scss]: Angular prod compilation doesn't like ```> div``` syntax, so need to see implications of removing or find resolution.

File: buildScripts/crypto-patch.js [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
+// fs.readFile(path, 'utf8', (err, data) => {
+//   if (err) {
+//     return console.log(err);
+//   }
+//   const result = data.replace(/node: false/g, 'node: {crypto: true, stream: true}');
+//
+//   fs.writeFile(path, result, 'utf8', (error) => {
+//     if (error) return console.log(error);
+//   });
+// });
```
Comment (andrewsetterfield) [inline buildScripts/crypto-patch.js]: did we mean to leave this commented out?

File: src/app/pages/journal/components/journal-navigation/journal-navigation.scss (hunk 2 above)
Comment (andrewsetterfield) [reply src/app/pages/journal/components/journal-navigation/journal-navigation.scss]: im not a fan of it either. should be simple enough to apply those margins in a better way

File: buildScripts/crypto-patch.js (hunk 3 above)
Comment (matthew2564) [reply buildScripts/crypto-patch.js]: Yes, wanted to see if it was needed any longer, left comment on top line suggesting we might be able to remove it if it causes no issues

File: src/app/pages/journal/components/journal-navigation/journal-navigation.scss (hunk 2 above)
Comment (matthew2564) [reply src/app/pages/journal/components/journal-navigation/journal-navigation.scss]: Removed as syntax incorrect and therefore the output is not evident

**Review by andrewsetterfield — APPROVED** [review]

**Review by gittins7 — APPROVED** [review]

**Review by solomonsimukoko — APPROVED** [review]

**Pull request merged**